### PR TITLE
Special Character Remover - PDI Plugin - Marketplace

### PR DESCRIPTION
**Pull request by rishuatgithub**
Hi This is a Pentaho Kettle Step Plugin which aims at improving the data quality of the input data stream. I would like this plugin to be available in the Pentaho Marketplace. Please kindly have a look at the file and let me know incase of any concerns.

[Note: This is a fresh PR. I have requested a PR earlier, which due to my mistakes got messed up. Please close the prev. request id #343 and consider this as a new one.]

Thanks :)

### CODE REVIEW

File: marketplace.xml [hunk 1]
```diff
@@ -7265,5 +7265,50 @@ Pentaho currently does not provide a mechanism allowing for plugin steps to be d
...
+  <support_message>Supported by rishu.shrivastava@gmail.com</support_message>
+  <support_organization>Rishu Shrivastava</support_organization>
+  
+  <versions>
+    <version>
+      <branch>Stable</branch>
+      <version>1.1.0</version>
+      <package_url>https://github.com/rishuatgithub/Special-Character-Remover/releases/download/SpecialCharacterRemover_v1.1.0/SpecialCharacterRemover-ver-1.1.0.zip</package_url>
+      <samples_url>https://github.com/rishuatgithub/Special-Character-Remover/releases/download/SpecialCharacterRemover_v1.1.0/SpecialCharacterRemover-ver-1.1.0.zip</samples_url>
+      <build_id>1.1.0<build_id>
```
Comment (mattyb149) [inline marketplace.xml]: This tag (build_id) isn't closed correctly.  To help avoid these problems, we have a schema validator for the XML file. Before committing, please run "ant" and make sure it runs successfully, thanks!

**Comment by rishuatgithub** [comment]
I have corrected the error in marketplace xml. Please kindly have a look.

**Comment by mattyb149** [comment]
My apologies @rishuatgithub, due to a bug in the Marketplace (which also shows up in http://jira.pentaho.com/browse/PDI-14017), your plugin will install correctly but not show up as installed when you go back to the Marketplace. This prevents someone from using the Marketplace to uninstall (although they can still uninstall manually by deleting your folder under plugins/steps/.

I can merge this PR as-is, since it is not your fault this bug exists. However if you wish to rename your ZIP (and/or root-level directory inside the ZIP) as described in the aforementioned Jira case, I will wait to hear from you before proceeding.

**Comment by rishuatgithub** [comment]
@mattyb149 is it OK if i can change the id  (inside the xml) to "SpecialCharacterRemover-ver-1.1.0" instead of changing the zip file? Will it behave the same as in the jira case.? Please suggest me.

**Comment by mattyb149** [comment]
That may fix the issue with the Marketplace not recognizing that it is installed, or it may fix the issue where it would uninstall (had it known it was installed), but probably not both :(

BTW, I usually test these PRs by fetching them into my local clone, and using a custom "marketplaces.xml" file to point at my local copy. That file goes in $HOME/.kettle/ and will be read by the Marketplace if it exists (otherwise it defaults to the master copy here in GitHub).  Here's a Gist with the contents:

https://gist.github.com/mattyb149/8bbe4b670c6dfb108a2b

For PRs like this, I do a "git checkout upstream/pr/349" then start PDI. It will read my marketplaces.xml and find your proposed addition. Then from PDI I test install, open the step dialog, and uninstall.  If you like, you can try the same approach, to see how your plugin will behave when it gets into the master marketplace. You don't have to checkout a PR of course, just use your local Git clone of marketplace-metadata, it will already have your proposed changes (if you're on that branch of course).

I will write up a blog on this procedure (on http://funpdi.blogspot.com) soon, please let me know if you have any questions or issues. Thanks!

**Comment by rishuatgithub** [comment]
@mattyb149 i tested the PR. Here are two scenarios faced in my case:

With difference in id and the zip file name, two separate download folders are getting created, one with the id and other with the zip file name. The marketplace in pdi was showing me "Installed" but when i try to "uninstall" it.. it is saying " the plugin has been uninstalled" but in reality its not.

Whereas, if i change the ID and ZIP file name as same, the plugin is getting installed properly into a single folder (same name as the zip), but when i try to uninstall it, it shows "unable to install".

So going by the JIRA case comment, i believe it would be better if i make both the ID and ZIP file names same. Your comments. Thanks !!

**Comment by mattyb149** [comment]
Yes I believe this will give the best results. Also if there is a root folder inside the ZIP, that should match the name and ID.

**Comment by rishuatgithub** [comment]
@mattyb149 i have updated the xml with the new id as tested earlier. Please check and if there is no roadblocks, then can we merge in the marketplace.

**Comment by rishuatgithub** [comment]
Thanks a lot Matt for getting this in the marketplace :)

**Comment by mattyb149** [comment]
My pleasure, thank you for your contribution!

**Pull request merged**